src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,29 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
```

Comment:
It's unnecessary to check if the session is null after calling `factory.openSession()`. The session should be initialized correctly by the factory, and if it fails, it will throw an exception.